tsdb/series_partition.go

```diff
@@ -309,6 +309,13 @@ func (p *SeriesPartition) DeleteSeriesID(id uint64) error {
 		return err
 	}
 
+	// Flush active segment write.
+	if segment := p.activeSegment(); segment != nil {
+		if err := segment.Flush(); err != nil {
+			return err
+		}
+	}
```

Comment:
This was an unrelated bug but it caused series deletions to not be visible in compactions until another series was added afterward which caused the segment to be flushed. Now all deletions cause a segment flush.